frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts

```diff
@@ -167,6 +178,260 @@ export const filterRevisionsByActiveApplication = (
+ * @param resource
+ * @param resources
+ */
+export const getPubSubSubscribers = (
+  resource: K8sResourceKind,
+  resources: TopologyDataResources,
+): Subscriber[] | [] => {
+  const channelResourceProps = getDynamicChannelModelRefs();
+
+  const relationShipMap = {
+    Broker: [
+      {
+        relatedResource: 'ksservices',
+        relationshipResource: 'triggers',
+        isRelatedResource: isSubscriber,
+      },
+    ],
+    Service: [
+      {
+        relatedResource: 'brokers',
+        relationshipResource: 'triggers',
+        isRelatedResource: isPublisher,
+      },
+    ],
+  };
+  _.forEach(channelResourceProps, (channel) => {
+    relationShipMap.Service.push({
+      relatedResource: channel,
+      relationshipResource: 'eventingsubscription',
+      isRelatedResource: isPublisher,
+    });
+    relationShipMap[channel] = [
+      {
+        relatedResource: 'ksservices',
+        relationshipResource: 'eventingsubscription',
+        isRelatedResource: isSubscriber,
+      },
+    ];
+  });
+
+  let subscribers = [];
+  if (relationShipMap[resource.kind] || relationShipMap[referenceFor(resource)]) {
+    const depicters = relationShipMap[resource.kind] || relationShipMap[referenceFor(resource)];
+    _.forEach(depicters, (depicter) => {
+      const { relatedResource, relationshipResource, isRelatedResource } = depicter;
+      if (resources[relatedResource] && resources[relatedResource].data.length > 0) {
+        subscribers = subscribers.concat(
+          _.reduce(
+            resources[relatedResource].data,
+            (acc, relRes) => {
+              if (isInternalResource(relRes) || !isRelatedResource) {
+                return acc;
+              }
+              const relationshipResources = (resources[relationshipResource].data || []).filter(
+                (relationshipRes) => {
+                  return isRelatedResource(relRes, relationshipRes, resource);
+                },
+              );
+              const relationShipData = relationshipResources.map((res) => {
+                return {
+                  kind: referenceFor(res),
+                  name: res.metadata.name,
+                  ...getTriggerFilters(res),
+                };
+              });
+              if (relationShipData.length > 0) {
+                const obj = {
+                  kind: referenceFor(relRes),
+                  name: relRes.metadata.name,
+                  data: relationShipData,
+                };
+                acc.push(obj);
+              }
+              return acc;
+            },
+            [],
+          ),
+        );
+      }
+    });
+  }
+  return subscribers;
+};
+
+/**
+ * Get the knative service subscriptions
+ * @param ksvc Knative Service
+ * @param resources
+ */
+export const getSubscribedPubSubNodes = (
+  ksvc: K8sResourceKind,
+  resources: TopologyDataResources,
+) => {
+  type test = {
+    channels: any;
+    brokers: string[];
+  };
```

Comment:
Proper name for this type and proper type for channels?

Reply:
added type